lib/Service/AntiAbuseService.php

```diff
@@ -85,11 +86,11 @@ private function checkNumberOfRecipients(IUser $user,
 			return;
 		}
 
-		$actualNumberOfRecipients = count($messageData->getTo())
-			+ count($messageData->getCc())
-			+ count($messageData->getBcc());
+		$actualNumberOfRecipients = count($message->getRecipients());
 
 		if ($actualNumberOfRecipients >= $numberOfRecipientsThreshold) {
+			$message->setStatus(LocalMessage::STATUS_TOO_MANY_RECIPIENTS);
+			$this->eventDispatcher->dispatchTyped(new OutboxMessageStatusChangeEvent($message));
```

Comment:
that is a lot of indirection. why is it necessary to dispatch an event here for a database update?